contracts/src/CollateralRegistry.sol

```diff
@@ -55,33 +38,68 @@ contract CollateralRegistry is ICollateralRegistry {
-        troveManager3 = numTokens > 3 ? _troveManagers[3] : ITroveManager(address(0));
-        troveManager4 = numTokens > 4 ? _troveManagers[4] : ITroveManager(address(0));
-        troveManager5 = numTokens > 5 ? _troveManagers[5] : ITroveManager(address(0));
-        troveManager6 = numTokens > 6 ? _troveManagers[6] : ITroveManager(address(0));
-        troveManager7 = numTokens > 7 ? _troveManagers[7] : ITroveManager(address(0));
-        troveManager8 = numTokens > 8 ? _troveManagers[8] : ITroveManager(address(0));
-        troveManager9 = numTokens > 9 ? _troveManagers[9] : ITroveManager(address(0));
+        for(uint8 i=0; i<numTokens; i++) {
+            collateralTokens[i] = _tokens[i];
+            troveManagers[i] = _troveManagers[i];
+        }
 
         // Initialize the baseRate state variable
         baseRate = INITIAL_BASE_RATE;
         emit BaseRateUpdated(INITIAL_BASE_RATE);
     }
 
+    function addNewCollaterals(uint256[] memory _indexes, IERC20Metadata[] memory _tokens, ITroveManager[] memory _troveManagers) external override {
+        require(msg.sender == boldToken.getOwner(), "Only owner");
+        
+        uint256 numTokens = _indexes.length; 
+        require(numTokens > 0 && numTokens == _tokens.length && numTokens == _troveManagers.length, "Invalid input");
+
+        require(totalCollaterals + numTokens <= 10, "Max collaterals");
+
+        // add new collaterals and trove managers 
+        for(uint8 i=0; i<numTokens; i++) {
+            uint256 index = _indexes[i];
+
+            // can't overwrite an existing branch
+            require(index <= 9, "Invalid index");
+            require(address(collateralTokens[index]) == address(0), "Collateral already initialised");
+
+            collateralTokens[index] = _tokens[i];
+            troveManagers[index] = _troveManagers[i];
+
+            emit NewCollateralAdded(address(_tokens[i]), address(_troveManagers[i]));
+        }
+
+        // update total Collaterals
+        totalCollaterals += numTokens;
+    }
+ 
+    function removeCollaterals(uint256[] memory _indexes) external override {
+        require(msg.sender == boldToken.getOwner(), "Only owner");
+       
+        uint256 numTokens = _indexes.length; 
+        require(
+            numTokens > 0 && 
+            numTokens <= totalCollaterals, 
+            "Invalid input"
+        );
+
+        // remove collaterals and trove managers 
+        for(uint8 i=0; i<numTokens; i++) {
+            uint256 index = _indexes[i];
+
+            require(index <= 9, "Invalid index");
+            require(address(collateralTokens[index]) != address(0), "Branch not initialised");
+
+            emit CollateralRemoved(address(collateralTokens[index]), address(troveManagers[index]));
+
+            collateralTokens[index] = IERC20Metadata(address(0));
+            troveManagers[index] = ITroveManager(address(0));
+        }
+        
+        totalCollaterals -= numTokens;
```

Comment:
i dont think this works. we always need to go over the whole list since some in the middle might be removed

Reply:
should we simply disable colalteral instead of removing? so people can still redeem etc. ?

Reply:
true omg